src/ng-select/ng-select.component.ts

```diff
@@ -486,20 +487,27 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
         }
     }
 
-    private _selectWriteValue(ngModel: any | any[]) {
-        if (!this._isDefined(ngModel)) {
+    private _handleWriteValue(ngModel: any | any[]) {
+        const isEmptyArray = ngModel && Array.isArray(ngModel) && ngModel.length === 0;
+        if (ngModel === null || isEmptyArray) {
+            this.itemsList.clearSelected();
             return;
         }
 
         const select = (val: any) => {
-            const item = this.itemsList.findItem(val);
+            let item = this.itemsList.findItem(val);
             if (item) {
                 this.itemsList.select(item);
             } else {
-                const isObject = val instanceof Object;
-                const simpleValue = !isObject && !this.bindValue;
-                if (isObject || simpleValue) {
+                const isValObject = isObject(val)
+                const isPrimitive = !isValObject && !this.bindValue;
+                if ((isValObject || isPrimitive)) {
                     this.itemsList.select(this.itemsList.mapItem(val, null));
+                } else if (this.bindValue && !this._isTypeahead) {
+                    item = {};
```

Comment:
```javascript 
item = {
    [this.bindLabel]: null,
    [this.bindValue]: val,
};
```

Reply:
Also I think we should support this use case for typeahead too. Users would need to preload items to see selected.
```javascript
else if (this.bindValue) {
```

Reply:
<img width="271" alt="screen shot 2018-03-07 at 16 18 08" src="https://user-images.githubusercontent.com/17824796/37097044-29cd25d2-2223-11e8-8da3-48f51c2e9d12.png">
on multiselect, label is visible